SweetCersei.js

```diff
@@ -0,0 +1,19 @@
+const DrawCard = require('../../drawcard.js');
+
+class SweetCersei extends DrawCard {
+    setupCardAbilities(ability) {
+        this.reaction({
+            when: {
+                afterChallenge: ({challenge}) => challenge.winner === this.controller && 'intrigue' && !this.controller.cannotGainGold
```

Comment:
Couple of comments:

- The ```player.cannotGainGold``` flag is used to switch on/off the effect of Devout Freerider, which is not applicable to the effect of this card.
- You want to check 'intrigue' vs. the challengeType of the challenge.
- Even though there's a bunch of them left in the code, we try to not deconstruct event objects anymore.

Putting these together, this should do the trick:

```javascript
afterChallenge: event => event.challenge.winner === this.controller && event.challenge.challengeType === 'intrigue'
```